Bottom: eb73f4803a40e64f1ec0442038cc09c6c8da9fda
Top:    c01d8f1307fcab04129b21f5ab037212755a5469
Author: Ian Kent <ikent@redhat.com>
Date:   2013-04-04 17:09:46 +0800

Refresh of autofs-5.0.7-syncronize-handle_mounts-shutdown.patch

---

diff --git a/daemon/automount.c b/daemon/automount.c
index cc23393..805128f 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -1477,8 +1477,9 @@ static void handle_mounts_finish(void)
 	pthread_setcancelstate(PTHREAD_CANCEL_DISABLE, &cancel_state);
 
 	finish_mutex_lock();
-	/* Poke signal handler */
-	pthread_kill(state_mach_thid, SIGTERM);
+	/* Poke signal handler if we haven't already done so */
+	if (!fc.busy)
+		pthread_kill(state_mach_thid, SIGTERM);
 	fc.busy++;
 	error(LOGOPT_ANY, "before signal fc.busy %d", fc.busy);
 	finish_cond_wait();
